src/main/java/com/mongodb/hibernate/internal/translate/SelectStatementMqlTranslator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate;
+
+import static com.mongodb.hibernate.internal.VisibleForTesting.AccessModifier.PRIVATE;
+import static com.mongodb.hibernate.internal.translate.AstVisitorValueDescriptor.COLLECTION_AGGREGATE;
+
+import com.mongodb.hibernate.internal.VisibleForTesting;
+import com.mongodb.hibernate.internal.translate.mongoast.command.AstCommand;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.query.spi.QueryOptions;
+import org.hibernate.sql.ast.tree.Statement;
+import org.hibernate.sql.ast.tree.select.SelectStatement;
+import org.hibernate.sql.exec.spi.JdbcOperationQuerySelect;
+import org.hibernate.sql.exec.spi.JdbcParameterBindings;
+import org.jspecify.annotations.Nullable;
+
+final class SelectStatementMqlTranslator extends AbstractMqlTranslator<JdbcOperationQuerySelect> {
+
+    private final SelectStatement selectStatement;
+
+    SelectStatementMqlTranslator(SessionFactoryImplementor sessionFactory, SelectStatement selectStatement) {
+        super(sessionFactory);
+        this.selectStatement = selectStatement;
+    }
+
+    @Override
+    public JdbcOperationQuerySelect translate(
+            @Nullable JdbcParameterBindings jdbcParameterBindings, QueryOptions queryOptions) {
+        var aggregateCommand = getAggregateCommand();
+        var mql = renderMongoAstNode(aggregateCommand);
+        var sessionFactory = getSessionFactory();
+        var jdbcValuesMappingProducer = sessionFactory
+                .getFastSessionServices()
```

Comment:
`FastSessionServices` is in the `internal` package, so we should not use it. I propose us to https://github.com/stIncMale/mongo-hibernate/commit/268bf5edaa1db6707c128251f5eab4de78ff4452, which is based on the code I proposed in https://github.com/mongodb/mongo-hibernate/pull/69#discussion_r2004584693.

Reply:
Good point! I like it! I adopted your code with only one minor change by replacing
```
when(sessionFactory.getServiceRegistry()).thenReturn(serviceRegistry);
```
with
```
doReturn(serviceRegistry).when(sessionFactory).getServiceRegistry();
```
to maintain mockito usage consistency. For some reason (I am not 100% clear), `doReturn()` is more robust than `when().then()` as well.